src/rotation/log.zig

```diff
@@ -7,65 +7,235 @@ const Neighbor = main.chunk.Neighbor;
-	_ = baseModel.transformModel(rotation.rotationMatrixTransform, .{Mat4f.rotationZ(std.math.pi/2.0).mul(Mat4f.rotationX(-std.math.pi/2.0))});
-	_ = baseModel.transformModel(rotation.rotationMatrixTransform, .{Mat4f.rotationX(-std.math.pi/2.0)});
-	_ = baseModel.transformModel(rotation.rotationMatrixTransform, .{Mat4f.rotationZ(std.math.pi).mul(Mat4f.rotationX(-std.math.pi/2.0))});
-	rotatedModels.put(modelId, modelIndex) catch unreachable;
-	return modelIndex;
+pub fn createBlockModel(_: Block, _: *u16, _: ZonElement) ModelIndex {
+	if(modelIndex) |idx| return idx;
+
+	for(0..64) |i| {
+		var quads = main.List(main.models.QuadInfo).init(main.stackAllocator);
+		defer quads.deinit();
+
+		const data = LogData.init(@intCast(i));
+
+		for(Neighbor.iterable) |neighbor| {
+			const pattern = getPattern(data, neighbor);
+
+			quads.append(rotateQuad(pattern, neighbor));
+		}
+
+		const index = main.models.Model.init(quads.items);
+		if(i == 0) {
+			modelIndex = index;
+		}
+	}
+
+	return modelIndex.?;
 }
 
 pub fn model(block: Block) ModelIndex {
-	return .{.index = blocks.meshes.modelIndexStart(block).index + @min(block.data, 5)};
+	return .{.index = blocks.meshes.modelIndexStart(block).index + (block.data & 63)};
 }
 
-pub fn rotateZ(data: u16, angle: Degrees) u16 {
-	comptime var rotationTable: [4][6]u8 = undefined;
-	comptime for(0..6) |i| {
-		rotationTable[0][i] = i;
-	};
-	comptime for(1..4) |a| {
-		for(0..6) |i| {
-			const neighbor: Neighbor = @enumFromInt(rotationTable[a - 1][i]);
-			rotationTable[a][i] = neighbor.rotateZ().toInt();
+pub const rotateZ = branch.rotateZ;
+
+pub fn generateData(
+	_: *main.game.World,
+	pos: Vec3i,
+	_: Vec3f,
+	_: Vec3f,
+	_: Vec3i,
+	neighbor: ?Neighbor,
+	currentBlock: *Block,
+	neighborBlock: Block,
+	blockPlacing: bool,
+) bool {
+	const canConnectToNeighbor = currentBlock.mode() == neighborBlock.mode() and currentBlock.modeData() == neighborBlock.modeData();
+
+	if(blockPlacing or canConnectToNeighbor or !neighborBlock.replacable()) {
+		const neighborModel = blocks.meshes.model(neighborBlock).model();
+
+		var currentData = LogData.init(currentBlock.data);
+		// Log block upon placement should extend towards a block it was placed
+		// on if the block is solid or also uses log model.
+		const targetVal = ((!neighborBlock.replacable() and (!neighborBlock.viewThrough() or canConnectToNeighbor)) and (canConnectToNeighbor or neighborModel.isNeighborOccluded[neighbor.?.reverse().toInt()]));
+		currentData.setConnection(neighbor.?, targetVal);
+
+		for(Neighbor.iterable) |side| {
+			if(side == neighbor.?) {
+				continue;
+			}
+
+			const sidePos = pos + side.relPos();
+			const sideBlock = main.renderer.mesh_storage.getBlock(sidePos[0], sidePos[1], sidePos[2]) orelse continue;
+			const canConnectToSide = currentBlock.mode() == sideBlock.mode() and currentBlock.modeData() == sideBlock.modeData();
```

Comment:
Why are you comparing the mode data? This rotation mode doesn't have any mode data.